### ISSUE TITLE
Jacobian point comparison doesn't seem to work

### ISSUE THREAD

**Issue by tuxxy** [paritytech/libsecp256k1#42]
I've noticed that comparing Jacobian points doesn't seem to work very well. I discovered this after trying to compare `jacobian_basepoint_g == scalar_one * jacobian_basepoint_g`.

The only way to get the comparison to succeed is to convert them to Affine via `Affine::from_gej` and then perform the comparison. It seems the library needs a proper Jacobian point comparison function implemented.

**Comment by sorpaas** [comment]
@tuxxy Yeah looks like the current `Eq` was a simple derive of `Jacobian` struct. It has a scale factor `z`, with different `x`s` and `y`s might still represent the same value. I removed the flawed derive.

For now converting Jacobian to Affine and then do the comparison should be the way to go. Let me see if there could be better more efficient ways.

**Comment by tuxxy** [comment]
@sorpaas [See here for a Solidity implementation of efficient Jacobian point comparison](https://github.com/nucypher/numerology/blob/master/contracts/Numerology.sol#L10) we wrote to use in our smart contracts. It should be helpful enough for your interest. :)
